docs/references/message-execution-properties.mdx

```diff
@@ -2,67 +2,55 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
 
 # Properties of message executions on ICP
 
-<MarkdownChipRow labels={["Reference"]} />
+<MarkdownChipRow labels={["Intermediate", "Reference", "Security"]} />
 
-There are a few key properties that one needs to understand about the execution model on ICP. Let's walk through them.
+## Asynchronous messaging model
 
-### Property 1
+ICP relies on an asynchronous messaging model. Compared to synchronous messaging like on Ethereum, this provides performance advantages because individual messages can be flexibly scheduled. However, asynchrounous message execution can also lead to sometimes unexpected or unintuitive behavior. Therefore, it is important to understand the properties of message execution. Potential security issues that arise in this model, such as reentrancy bugs, are discussed in the [security best practices on inter-canister calls](/docs/current/developer-docs/security/security-best-practices/inter-canister-calls).  
 
-Message execution is sequential, i.e. only a single message per canister is processed at a time.
+The [community conversation on security best practices](https://www.youtube.com/watch?v=PneRzDmf_Xw&list=PLuhDt1vhGcrez-f3I0_hvbwGZHZzkZ7Ng&index=2&t=4s) also discusses the messaging properties.
 
+## Message execution properties
 
-### Property 2
+A **call** is a canister's implementation of either an [update](/references/ic-interface-spec.md#http-call) or [query call](/docs/current/references/ic-interface-spec#http-query) that it exposes. For example, if the Rust CDK is used, these are usually annotated with `#[query]` or `#[update]`, respectively. ICP also supports additional [entry points](/references/ic-interface-spec.md#entry-points) such as heartbeats, timers, initialization or upgrade hooks. A **message** is a set of consecutive instructions that a subnet executes for a canister. The code execution for such an entry point can be split into several messages if inter-canister calls are made. The following properties are essential:
```

Comment:
```suggestion
To interact with a canister's methods, there are two primary types of **calls** that can be used: [update](/references/ic-interface-spec.md#http-call) calls and [query](/docs/current/references/ic-interface-spec#http-query) calls.
If the Rust CDK is used, these are usually annotated with `#[update]` or `#[query]`, respectively. ICP also supports additional [entry points](/references/ic-interface-spec.md#entry-points) such as heartbeats, timers, initialization or upgrade hooks. 

A **message** is a set of consecutive instructions that a subnet executes for a canister. The code execution for such any entry point can be split into several messages if inter-canister calls are made. The following properties are essential:
```

This is the same explanation as https://internetcomputer.org/docs/current/developer-docs/smart-contracts/call/overview which I think is more accurate to describe what a call is.

Reply:
Are we missing the property that message execution is atomic?

Reply:
Not to conflict with Roel's suggestion:
- "code execution for such any entry point" -> "any such"
- say how updates are represented in Motoko, since we're using a Motoko example below

Reply:
I'd add that it is also atomic which is not necessary for messages that are executed sequential. Messages could be executed in sequence but partially.

Reply:
Maybe that is property 3?

Reply:
Ok that works, adapted Property 1 as follows: 

> Message execution is atomic and sequential, and never parallel.